core/jobs.py

```diff
@@ -195,12 +195,15 @@ def register_start(cls, job_id, metadata=None):
         cls._post_start_hook(job_id)
 
     @classmethod
-    def register_completion(cls, job_id, output_list):
+    def register_completion(
+            cls, job_id, output_list, max_output_len_chars=None):
         """Marks a job as completed.
 
         Args:
             job_id: str. The ID of the job to complete.
             output_list: list(object). The output produced by the job.
+            max_output_len_chars: int or None. Overrides the intended max
```

Comment:
Write this docstring so that it "faces externally" and describes an interface -- i.e. callers of this function should know what to pass in here based on the docstring. You're referencing an internal function, which is an implementation detail and doesn't really give insight into what value to select. Also, there should be a description about what None signifies.